### DISCUSSION TITLE
Error in DataPortal Fetch with net framework 4.0

### DISCUSSION THREAD

**Discussion by ProDInfo** [MarimerLLC/csla#2214] [Questions]
**Describe the bug**
When use DataPortal.Fetch with net framework 4.0 not work

Await DataPortal.FetchAsync(Of ...)

Give a error that DataPortal_Create is not implemented, but it's implemnted with CreateAttribute

**Version and Platform**
CSLA version: 5.4.1
OS: Windows
Platform: WinForms, WPF

**Code that Fails**

        <Create>
        Protected Sub Create()
            NotAuthenticated()
        End Sub

If I use below code then works.

        <Create>
        Protected Sub DataPortal_Create()
            NotAuthenticated()
        End Sub
**Stack Trace or Exception Detail**
DataPortal.Create falló (DataPortal_Create no implementado)"

**Additional context**
The problem is on net framework 4.0 that don't work fetchattribute properly

**Comment by rockfordlhotka** [comment]
In some CSLA base classes there is a default `DataPortal_Create` method that you can choose to override. I don't think there are any base classes with a default `DataPortal_Fetch`, so you always have to implement your own.

This will be changing in CSLA 6, because we are removing all the base methods, so you always have to implement the ones you want explicitly.

**Reply by ProDInfo** [reply]
The same code work on net framework 4.8, but not on 4.0

**Reply by rockfordlhotka** [reply]
fwiw, we're dropping support for .NET 4.0 in CSLA 6 (later this year), so this is a short-term issue.

What base class are you using in your class? BusinessBase, ReadonlyBase, BusinessListBase?

**Reply by ProDInfo** [reply]
I inherit from CslaIdentityBase

**Reply by rockfordlhotka** [reply]
I'll try that base class when I get a chance.

You should know that Microsoft is generally not supporting custom principal/identity types in .NET 5 or 6. Technically they still kind of work, but some environments (like aspnetcore and Blazor) insist on using standard `ClaimsPrincipal` and `ClaimsIdentity` types. As a result, the `CslaIdentityBase` and related types are slowly going away, because they simply don't work anymore in the latest versions of .NET.

**Reply by rockfordlhotka** [reply]
I am able to reproduce the issue.

I'm not sure it is worth the effort to fix, given that

1. you have a workaround
2. CSLA 5.5.0 will be the last version supporting .NET 4; I'm finally following Microsoft's lead and discontinuing support for unsupported versions of .NET

That said, if you want to track down the problem and submit a PR prior to the v5.5 release I'll accept it.
